python/ray/train/tests/test_gpu.py

```diff
@@ -171,13 +184,13 @@ def train_fn():
         # Make sure model is on cuda.
         assert next(model.parameters()).is_cuda
 
-    trainer = Trainer("torch", num_workers=2, use_gpu=True)
-    trainer.start()
-    trainer.run(train_fn)
-    trainer.shutdown()
+    trainer = TorchTrainer(
+        train_fn, scaling_config=ScalingConfig(num_workers=2, use_gpu=True)
+    )
+    results = trainer.fit()
+    assert not results.error
```

Comment:
is this necessary? any errors should immediately be propagated up hopefully...if not then we definitely need to fix that